airflow/cli/cli_parser.py

```diff
@@ -65,6 +65,22 @@
 
 ALL_COMMANDS_DICT: dict[str, CLICommand] = {sp.name: sp for sp in airflow_commands}
 
+# Check if sub-commands are defined twice, which could be an issue.
+if len(ALL_COMMANDS_DICT) < len(airflow_commands):
+    seen = set()
+    dup = []
+    for command in airflow_commands:
+        if command.name not in seen:
+            seen.add(command.name)
+        else:
+            dup.append(command.name)
+    log.warning(
+        "The following CLI %d command(s) are defined more than once, "
+        "CLI behavior when using those will be unpredictable: %s",
+        len(dup),
+        dup,
+    )
```

Comment:
I would do something like

```python
import collection

dups = {k for k, v in collection.Counter(airflow_commands).items() if v > 1}
log.warning(
    "The following CLI command(s) are defined more than once, "
    "CLI behavior when using those will be unpredictable: %s",
    sorted(dups),
)
```

Reply:
it's more compact but I personally find it less readable.
(it's also less efficient but that does not matter here)

Reply:
Opinion: I actually like the compact way better :). I learned to appreciate some of the itertools/collections shorthands, this one looks pretty readable.

IMHO. I think for the compactness of some of the comprehensions and shorthands like that, the optimisation should be for readability, not "length" (I recall the programming competitiions in writing something as short as possible, usually those winners were also winning the "least readable" code competition as well). But this one is both more compact and pretty readable as well.

Reply:
actually it has to be `{k for k, v in collections.Counter([c.name for c in airflow_commands]).items() if v > 1}`
I'll push it in a new commit, battling with tests at the moment.